component/src/main/java/org/cbioportal/genome_nexus/util/GenomicVariantUtil.java

```diff
@@ -14,35 +12,70 @@
 public class GenomicVariantUtil {
 
     public static GenomicVariant fromHgvs(String hgvs) {
-        if (!isHgvs(hgvs)) {
-            throw new InvalidHgvsException();
-        }
-        String chr = getPattern("^.+(?=:)", hgvs);
-        RefType ref_type = getRefTypeFromHgvs(hgvs);
-        Integer start = Integer.parseInt(getPattern("(?<=\\.)\\d+(?=[_ATGC])", hgvs));
+        validateHgvsRefType(hgvs);
+        // check chromosome reference is valid
+        String chr = getChrFromHgvs(hgvs);
+        Integer start = getStartFromHgvs(hgvs);
         Integer end = getEndFromHgvs(hgvs, start);
-        Type type = getTypeFromHgvs(hgvs);
+        GenomicVariant.Type type = getTypeFromHgvs(hgvs);
+        if (type == GenomicVariant.Type.INSERTION && end != start + 1) {
+            throw new InvalidHgvsException("insertion encountered where end position did not equal start position plus 1");
+        }
         String ref = getRefFromHgvs(hgvs, type);
         String alt = getAltFromHgvs(hgvs, type);
-        return new GenomicVariant(chr, ref_type, start, end, type, ref, alt);
+        return new GenomicVariant(chr, start, end, type, ref, alt);
+    }
+
+    private static void validateHgvsRefType(String hgvs) {
+        String ref_type = getPattern("(?<=:)[cgmrp](?=.)", hgvs);
+        if (ref_type == null || ref_type.trim().length() == 0) {
+            throw new InvalidHgvsException();
+        }
+        if (!ref_type.trim().equals("g")) {
+            throw new RefTypeNotSupportedException();
+        }
     }
 
+/* The following (fromRegion) code is invalid
```

Comment:
```suggestion
/* TODO: The following (fromRegion) code is invalid
```
I'm guessing the idea behind this comment is that it is a TODO? Like we should fix it, but we don't do it as part of this PR?

Reply:
Yes, I created separate issue #402 for fixing this